### PR TITLE
Added GRANT, REVOKE, and the remainder of CREATE USER

### PR DESCRIPTION
**Pull request by Hydrocharged**

### CODE REVIEW

File: go/vt/sqlparser/sql.y [hunk 1]
```diff
@@ -856,6 +890,358 @@ default_role_opt:
...
+  }
+
+user_comment_attribute:
+  {
+    $$ = ""
+  }
+| COMMENT_KEYWORD STRING
+  {
+    comment := string($2)
+    $$ = `{"comment": "`+escapeDoubleQuotes(comment)+`"}`
```
Comment (zachmu) [inline go/vt/sqlparser/sql.y:1059]: Why a JSON object here instead of just a string?

File: go/vt/sqlparser/sql.y [hunk 2]
```diff
@@ -856,6 +890,358 @@ default_role_opt:
...
+    allPriv := []Privilege{Privilege{Type: PrivilegeType_All, Columns: nil}}
+    $$ = &RevokePrivilege{Privileges: allPriv, ObjectType: $4, PrivilegeLevel: $5, From: $7}
+  }
+| REVOKE grant_privilege_list ON grant_object_type grant_privilege_level FROM account_name_list
+  {
+    $$ = &RevokePrivilege{Privileges: $2, ObjectType: $4, PrivilegeLevel: $5, From: $7}
+  }
+| REVOKE ALL ',' GRANT OPTION FROM account_name_list
+  {
+    $$ = &RevokeAllPrivileges{From: $7}
```
Comment (zachmu) [inline go/vt/sqlparser/sql.y:1097]: Thoughts on having an `All` field in `RevokePrivilege` instead? In general, better to have a single AST node encapsulate a single logical action

Comment (zachmu) [reply go/vt/sqlparser/sql.y:1097]: Plus it looks like `Grant` already works like this

File: go/vt/sqlparser/ast_permissions.go [hunk 3]
```diff
@@ -388,6 +672,264 @@ func (d *DropRole) walkSubtree(visit Visit) error {
...
+		buf.Myprintf(" %s", user.String())
+	}
+	if g.WithAdminOption {
+		buf.Myprintf(" with admin option")
+	}
+}
+
+// walkSubtree implements the interface Statement.
+func (g *GrantRole) walkSubtree(visit Visit) error {
+	return nil
```
Comment (zachmu) [inline go/vt/sqlparser/ast_permissions.go:765]: We don't actually use this anywhere, want to remove it from the interface?

Comment (Hydrocharged) [reply go/vt/sqlparser/ast_permissions.go:765]: Yep! I've always wanted to delete this lol

File: go/vt/sqlparser/sql.y (hunk 1 above)
Comment (zachmu) [reply go/vt/sqlparser/sql.y:1059]: OK, that's appropriate in that case.

File: go/vt/sqlparser/sql.y (hunk 2 above)
Comment (zachmu) [reply go/vt/sqlparser/sql.y:1097]: K

**Pull request merged**